src/event-bridge/event-bus.ts

```diff
@@ -107,6 +123,29 @@ export interface IEventBus<E extends EventBusRuleInput = EventBusRuleInput>
    * Put one or more events on an Event Bus.
    */
   (event: Partial<E>, ...events: Partial<E>[]): void;
+
+  /**
+   * Creates a rule that matches all events on the bus.
+   *
+   * When no scope or id are given, the bus is used as the scope and the id will be `whenAny`.
+   * The rule created will be a singleton.
+   *
+   * When scope and id are given, a new rule will be created each time.
+   *
+   * Like all functionless, a rule is only created when the rule is used with `.pipe` or the rule is retrieved using `.rule`.
+   *
+   * ```ts
+   * const bus = new EventBus(scope, 'bus');
+   * const func = new Function(scope, 'func', async (payload: {id: string}) => console.log(payload.id));
+   *
+   * bus
+   *  .whenAny()
+   *  .map(event => ({id: event.id}))
+   *  .pipe(func);
+   * ```
+   */
+  whenAny(): EventBusPredicateRuleBase<E>;
```

Comment:
out of curiosity why did we call it `when` instead of `filter`? `whenAny` makes sense in this context but it could be `forEach` otherwise

Reply:
good question... do we want to treat broadcasters like lists?

my original thought was that it would be confusing to model as an array when you have to filter then map then send. 

`forEach` doesn't make sense to me until it was just a 1:1 send to `pipe` though `pipe` could be replaced with `forEach`.

1. List
   2. `filter().filter.().map().forEach()`
   3. `all().filter().map().forEach()`
2. Stream
   3. `when().map().pipe()`
   4. `whenAll().when().map().pipe()`

Reply:
`forEach` and `filter` doesn't imply a list, does it? `.forEach` and `.map` and `.filter` can be applied to lists, streams, whatever.

Reply:
I would rename `whenAny` or `whenAll` to `all`